server/controllers/bookingControllers.js

```diff
@@ -1,184 +1,145 @@
+    const book = await client.query('SELECT * FROM bookings WHERE trip_id=$1 AND user_id=$2', [
+      tripid, req.user.id,
     ]);
-    if (recordbooking ) {
-      // ###update trip seats
-        let updatetrip = client.query('UPDATE trips SET seating_capacity=$1 where id = $2',[
-          seating_capacity-1,findtripid.rows[0].id,
-        ])
 
-        const book= {
+    if (!book.rows[0]) {
+    // ##### inserting in table
+      const cols = 'booking_date, first_name,last_name, phonenumber, user_email, bus_license,origin,destination,trip_date,time,fare,trip_id,user_id';
+      const cels = '$1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13';
+      const data = [
+        moment().format(), first_name, last_name, phonenumber, email, bus_license_number, origin, destination, trip_date, time, fare, id, userid,
+      ];
+      const recordbooking = client.query(
+        `INSERT INTO bookings(${cols})VALUES(${cels})`, data,
+      );
+      if (recordbooking) {
+      // ###update trip seats
+        const updatetrip = client.query('UPDATE trips SET seating_capacity=$1 where id = $2', [
+          seating_capacity - 1, findtripid.rows[0].id,
+        ]);
+        if (!updatetrip) { return response.response(res, 500, 'error', 'Error running query!', true); }
+        const getbook = {
           booking_date: moment().format(),
-          first_name: firstname.toUpperCase(),
-          last_name: lastname.toUpperCase(),
-          phonenumber: userphone,
-          user_email: useremail.toLowerCase(),
-          bus_license_number: buslicence.toUpperCase(),
-          origin: origin.toUpperCase(),
-          destination: destination.toUpperCase(),
-          trip_date: tripdate,
-          trip_time: triptime,
-          farer:fare,
-        }
-
-
-        return response.response(res, 201,'success', book, false);
-  
+          first_name,
+          last_name,
+          phonenumber,
+          email,
+          bus_license_number,
+          origin,
+          destination,
+          trip_date,
+          time,
+          fare,
+        };
+        return response.response(res, 201, 'success', getbook, false);
+      }
+    } else {
+      return response.response(res, 409, 'error', 'booking already made!', true);
     }
   }
 
-else {
-    return response.response(res, 409,'error', 'booking already made!', true);
-  }
-};
 
+  static async getbookings(req, res) {
+  // ###Display all bookings made by users //as admin
+    if (req.user.is_admin) {
+      client.query('SELECT * FROM bookings', (err, result) => {
+        if (err) {
+          return response.response(res, 404, 'error', 'Error running query');
+        }
+        const resul = result.rows;
+        return response.response(res, 200, 'success', resul, false);
+      });
+    }
+    // ###Display bookings  user for user only
+    else {
+      const finduserid = await client.query('SELECT * FROM bookings WHERE user_id=$1', [
+        req.user.id,
```

Comment:
make use of object destructuring for objects where you are getting 2 or more variables from
```
const { id, is_admin } = req.user